### PR TITLE
Issue148 - Remove support of v8 and v9

### PR DESCRIPTION
**Pull request by gquerret**

### CODE REVIEW

**Comment by gquerret** [comment]
See issue #148

File: src/progress/web/method/cgidefs.i [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+   and Debugging. */
+DEFINE NEW GLOBAL SHARED VARIABLE debugging-enabled AS LOGICAL NO-UNDO
+    INITIAL TRUE.
+
+/* List of Misc. variables above.  Used for dumping out all misc. values in
+   debugging mode such as with printval.p. */
+DEFINE NEW GLOBAL SHARED VARIABLE MiscVarList     AS char NO-UNDO 
+    INITIAL "AppProgram,AppURL,HostURL,SelDelim,SelfURL,":U.
+
+&GLOBAL-DEFINE OUT-STREAM PUT {&WEBSTREAM} UNFORMATTED
```
Comment (jakejustus) [inline src/progress/web/method/cgidefs.i:163]: MAJOR From src/progress/pct/pctWSComp.p - Preprocessor variable out-stream should be SCOPED-DEFINED [rule](http://sonar.riverside-software.fr/coding_rules#rule_key=rssw-oe-main%3Aeu.rssw.antlr.proparse.checks.GlobalDefine)

File: src/progress/web/method/cgidefs.i [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+/* OUT-FILE is the same as OUT-STREAM, for now.  */
+&GLOBAL-DEFINE OUT-FILE {&OUT-STREAM}
```
Comment (jakejustus) [inline src/progress/web/method/cgidefs.i:165]: MAJOR From src/progress/pct/pctWSComp.p - Preprocessor variable out-file should be SCOPED-DEFINED [rule](http://sonar.riverside-software.fr/coding_rules#rule_key=rssw-oe-main%3Aeu.rssw.antlr.proparse.checks.GlobalDefine)

File: src/progress/web/method/cgidefs.i [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+&GLOBAL-DEFINE OUT {&OUT-STREAM}
```
Comment (jakejustus) [inline src/progress/web/method/cgidefs.i:166]: MAJOR From src/progress/pct/pctWSComp.p - Preprocessor variable out should be SCOPED-DEFINED [rule](http://sonar.riverside-software.fr/coding_rules#rule_key=rssw-oe-main%3Aeu.rssw.antlr.proparse.checks.GlobalDefine)

File: src/progress/web/method/cgidefs.i [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+&GLOBAL-DEFINE OUT-LONG EXPORT {&WEBSTREAM}
```
Comment (jakejustus) [inline src/progress/web/method/cgidefs.i:167]: MAJOR From src/progress/pct/pctWSComp.p - Preprocessor variable out-long should be SCOPED-DEFINED [rule](http://sonar.riverside-software.fr/coding_rules#rule_key=rssw-oe-main%3Aeu.rssw.antlr.proparse.checks.GlobalDefine)

File: src/progress/web/method/cgidefs.i [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+&GLOBAL-DEFINE OUT-FMT PUT {&WEBSTREAM}
```
Comment (jakejustus) [inline src/progress/web/method/cgidefs.i:168]: MAJOR From src/progress/pct/pctWSComp.p - Preprocessor variable out-fmt should be SCOPED-DEFINED [rule](http://sonar.riverside-software.fr/coding_rules#rule_key=rssw-oe-main%3Aeu.rssw.antlr.proparse.checks.GlobalDefine)

File: src/progress/web/method/cgidefs.i [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+&GLOBAL-DEFINE DISPLAY DISPLAY {&WEBSTREAM}
```
Comment (jakejustus) [inline src/progress/web/method/cgidefs.i:169]: MAJOR From src/progress/pct/pctWSComp.p - Preprocessor variable display should be SCOPED-DEFINED [rule](http://sonar.riverside-software.fr/coding_rules#rule_key=rssw-oe-main%3Aeu.rssw.antlr.proparse.checks.GlobalDefine)

File: src/progress/web/method/cgidefs.i [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+&GLOBAL-DEFINE END .
```
Comment (jakejustus) [inline src/progress/web/method/cgidefs.i:170]: MAJOR From src/progress/pct/pctWSComp.p - Preprocessor variable end should be SCOPED-DEFINED [rule](http://sonar.riverside-software.fr/coding_rules#rule_key=rssw-oe-main%3Aeu.rssw.antlr.proparse.checks.GlobalDefine)

File: src/progress/web/method/cgidefs.i [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+
+Author: B.Burton
+
+Created: 05/14/96
+
+---------------------------------------------------------------------------*/
+
+/* Stream where we write all web output */
+&IF DEFINED(WEBSTREAM) = 0 &THEN
+  &GLOBAL-DEFINE WEBSTREAM STREAM WebStream
```
Comment (jakejustus) [inline src/progress/web/method/cgidefs.i:30]: MAJOR From src/progress/pct/pctWSComp.p - Preprocessor variable webstream should be SCOPED-DEFINED [rule](http://sonar.riverside-software.fr/coding_rules#rule_key=rssw-oe-main%3Aeu.rssw.antlr.proparse.checks.GlobalDefine)

File: src/progress/pct/pctWSComp.p [hunk 9]
```diff
@@ -97,7 +97,7 @@ ASSIGN cOptions = cOptions + (IF cOptions EQ '':U THEN '':U ELSE ',':U) + (IF cD
 ASSIGN cOptions = cOptions + (IF cOptions EQ '':U THEN '':U ELSE ',':U) + (IF cWebObj THEN 'web-object':U ELSE 'include':U).
```
Comment (jakejustus) [inline src/progress/pct/pctWSComp.p:97]: MINOR Line is 124 characters long, exceeding maximum value of 120 [rule](http://sonar.riverside-software.fr/coding_rules#rule_key=rssw-oe-main%3Aeu.rssw.antlr.proparse.checks.LinesTooLong)

File: src/progress/pct/pctWSComp.p [hunk 10]
```diff
@@ -97,7 +97,7 @@ ASSIGN cOptions = cOptions + (IF cOptions EQ '':U THEN '':U ELSE ',':U) + (IF cD
...
 ASSIGN cOptions = cOptions + (IF cOptions EQ '':U THEN '':U ELSE ',':U) + (IF cKeepMCT THEN 'keep-meta-content-type':U ELSE '':U).
```
Comment (jakejustus) [inline src/progress/pct/pctWSComp.p:98]: MINOR Line is 130 characters long, exceeding maximum value of 120 [rule](http://sonar.riverside-software.fr/coding_rules#rule_key=rssw-oe-main%3Aeu.rssw.antlr.proparse.checks.LinesTooLong)

File: src/progress/web/method/session.i [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+ Updated: 04/04/98 pdigre@progress.com
+            Initial version
+          04/26/01 adams@progress.com
+            WebSpeed integration
+          05/23/01 mbaker@progress.com
+            Dynamics integration
+---------------------------------------------------------------------*/
+
+&IF DEFINED(session-included) = 0 &THEN
+&GLOBAL-DEFINE session-included YES
```
Comment (jakejustus) [inline src/progress/web/method/session.i:20]: MAJOR From src/progress/pct/pctWSComp.p - Preprocessor variable session-included should be SCOPED-DEFINED [rule](http://sonar.riverside-software.fr/coding_rules#rule_key=rssw-oe-main%3Aeu.rssw.antlr.proparse.checks.GlobalDefine)

File: src/progress/web/method/proto.i [hunk 12]
```diff
@@ -0,0 +1,162 @@
...
+
+    Syntax      : {src/web/method/proto.i}
+    Author(s)   : B.Burton 
+    Created     : 02/03/97
+    Notes       : This file is included by cgidefs.i.  
+  ------------------------------------------------------------------------*/
+
+/* Make sure this file is only included once */
+&IF DEFINED(PROTO_I) = 0 &THEN
+&GLOBAL-DEFINE PROTO_I TRUE
```
Comment (jakejustus) [inline src/progress/web/method/proto.i:19]: MAJOR From src/progress/pct/pctWSComp.p - Preprocessor variable proto_i should be SCOPED-DEFINED [rule](http://sonar.riverside-software.fr/coding_rules#rule_key=rssw-oe-main%3Aeu.rssw.antlr.proparse.checks.GlobalDefine)

**Comment by jakejustus** [comment]
SonarQube analysis reported 28 issues
- CRITICAL 1 critical
- MAJOR 12 major
- MINOR 15 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:
1. CRITICAL [pctWSComp.p#L46](https://github.com/Riverside-Software/pct/blob/3cbe2fd0a191ec9fdefb86fa5f4a365e47c62dd2/src/progress/pct/pctWSComp.p#L46): Unused variable BaseName [rule](http://sonar.riverside-software.fr/coding_rules#rule_key=rssw-oe-main%3Aeu.rssw.antlr.proparse.checks.UnusedVariable)
2. MAJOR [pctWSComp.p#L39](https://github.com/Riverside-Software/pct/blob/3cbe2fd0a191ec9fdefb86fa5f4a365e47c62dd2/src/progress/pct/pctWSComp.p#L39): Unused variable ForceComp [rule](http://sonar.riverside-software.fr/coding_rules#rule_key=rssw-oe-main%3Aeu.rssw.antlr.proparse.checks.VariableAssignedButNotRead)
3. MAJOR [pctWSComp.p#L113](https://github.com/Riverside-Software/pct/blob/3cbe2fd0a191ec9fdefb86fa5f4a365e47c62dd2/src/progress/pct/pctWSComp.p#L113): SUBSTRING function without 'type' attribute' [rule](http://sonar.riverside-software.fr/coding_rules#rule_key=rssw-oe-main%3Aeu.rssw.antlr.proparse.checks.I18NLength)
4. MINOR [pctWSComp.p#L60](https://github.com/Riverside-Software/pct/blob/3cbe2fd0a191ec9fdefb86fa5f4a365e47c62dd2/src/progress/pct/pctWSComp.p#L60): No :T or :U attribute in string [rule](http://sonar.riverside-software.fr/coding_rules#rule_key=rssw-oe-main%3Aeu.rssw.antlr.proparse.checks.StringAttribute)
5. MINOR [pctWSComp.p#L62](https://github.com/Riverside-Software/pct/blob/3cbe2fd0a191ec9fdefb86fa5f4a365e47c62dd2/src/progress/pct/pctWSComp.p#L62): No :T or :U attribute in string [rule](http://sonar.riverside-software.fr/coding_rules#rule_key=rssw-oe-main%3Aeu.rssw.antlr.proparse.checks.StringAttribute)
6. MINOR [pctWSComp.p#L84](https://github.com/Riverside-Software/pct/blob/3cbe2fd0a191ec9fdefb86fa5f4a365e47c62dd2/src/progress/pct/pctWSComp.p#L84): No :T or :U attribute in string [rule](http://sonar.riverside-software.fr/coding_rules#rule_key=rssw-oe-main%3Aeu.rssw.antlr.proparse.checks.StringAttribute)
7. MINOR [pctWSComp.p#L91](https://github.com/Riverside-Software/pct/blob/3cbe2fd0a191ec9fdefb86fa5f4a365e47c62dd2/src/progress/pct/pctWSComp.p#L91): No :T or :U attribute in string [rule](http://sonar.riverside-software.fr/coding_rules#rule_key=rssw-oe-main%3Aeu.rssw.antlr.proparse.checks.StringAttribute)
8. MINOR [pctWSComp.p#L93](https://github.com/Riverside-Software/pct/blob/3cbe2fd0a191ec9fdefb86fa5f4a365e47c62dd2/src/progress/pct/pctWSComp.p#L93): No :T or :U attribute in string [rule](http://sonar.riverside-software.fr/coding_rules#rule_key=rssw-oe-main%3Aeu.rssw.antlr.proparse.checks.StringAttribute)
9. MINOR [pctWSComp.p#L130](https://github.com/Riverside-Software/pct/blob/3cbe2fd0a191ec9fdefb86fa5f4a365e47c62dd2/src/progress/pct/pctWSComp.p#L130): Statement LEAVE should be on a separate line [rule](http://sonar.riverside-software.fr/coding_rules#rule_key=rssw-oe-main%3Aeu.rssw.antlr.proparse.checks.OneStatementPerLine)
10. MINOR [pctWSComp.p#L136](https://github.com/Riverside-Software/pct/blob/3cbe2fd0a191ec9fdefb86fa5f4a365e47c62dd2/src/progress/pct/pctWSComp.p#L136): No :T or :U attribute in string [rule](http://sonar.riverside-software.fr/coding_rules#rule_key=rssw-oe-main%3Aeu.rssw.antlr.proparse.checks.StringAttribute)

**Pull request merged**